libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -76,11 +119,31 @@ pair<string, string> IRGenerator::run(
 	return {warning + ir, warning + asmStack.print()};
 }
 
+void IRGenerator::verifyCallGraphs(
+	FunctionCallGraphBuilder::ContractCallGraph const& _creationGraph,
+	FunctionCallGraphBuilder::ContractCallGraph const& _deployedGraph
+)
+{
+	// m_creationFunctionList and m_deployedFunctionList are not used for any other purpose so
+	// we can just destroy them without bothering to make a copy.
+
+	verifyCallGraph(collectReachableCallables(_creationGraph), move(m_creationFunctionList));
+	m_creationFunctionList = {};
+
+	verifyCallGraph(collectReachableCallables(_deployedGraph), move(m_deployedFunctionList));
+	m_deployedFunctionList = {};
+}
+
 string IRGenerator::generate(
 	ContractDefinition const& _contract,
 	map<ContractDefinition const*, string_view const> const& _otherYulSources
 )
 {
+	// Remember to call verifyCallGraphs() (which clears the list of generated functions) if you
+	// want to reuse the generator.
+	solAssert(m_creationFunctionList.empty(), "");
```

Comment:
I'm pretty sure these can be local variables in this function - can't they?

Reply:
They only reason they're not is that they're used by `verifyCallGraphs()` which gets called separately by `CompilerStack`.

We could make them local if we made `generate()` call `verifyCallGraphs()` automatically. We would then also have to pass the graphs from `CompilerStack` either via `generate()` or `IRGenerator` constructor or the context.

Reply:
If they are in the annotations, then we can just call it here, can't we? It's not too elegant to call that function from the compilerstack anyway, I think.

Reply:
Sure. But as I said, I'll be making a separate PR to change them to annotations. In that case I'll also change this to use locals that that other PR.

Reply:
ok!

Reply:
Done in #10990.